usr/libexec/playtron/hardware-test-tool

```diff
@@ -391,6 +392,87 @@ class DisplayTest(PlaceholderTest):
             self.done = True
 
 
+class SpeakersTest(PlaceholderTest):
+    def __init__(self):
+        self.name = 'Speakers|扬声器'
+        self.reset()
+        self.start_time = None
+        self.time_count = 0
+        self.wav_file_name = '/tmp/test_record.wav'
+        self.msg_headphones_hint = 'Unplug headphones|拔下耳机'
+        self.audio_output_started = False
+        self.audio_input_started = False
+        self.cleanup_started = False
+
+    def start(self, screen, font):
+        self.reset()
+        self.start_time = time.perf_counter()
+
+    def audio_output(self,):
+        # 5.7 seconds is the exact amount of time for the "Front right. Front left." voice line to happen twice.
+        os.popen('timeout 5.7s speaker-test -c 2 -t wav')
+
+    def audio_input_record(self):
+        # Record from the microphone for 6 seconds.
+        subprocess.Popen('arecord -f cd -d 6 ' + self.wav_file_name, shell=True)
+
+    def audio_input_playback(self):
+        os.popen('aplay ' + self.wav_file_name)
+
+    def cleanup(self):
+        try:
+            os.remove(self.wav_file_name)
+        except:
+            print('File does not exist: ' + self.wav_file_name)
+
+    def draw(self, screen, font):
+        if time.perf_counter() - self.start_time > 1:
+            self.start(screen, font)
+            self.time_count += 1
```

Comment:
This seems unnecessary, you are just counting the time again.
I would do away with this condition and just do
`elapsed_time = time.perf_counter() - self.start_time` and use `elapsed_time` instead of `time_count`

Reply:
Good point, thanks! I'll refactor that.

Reply:
I found the logic with `elapsed_time = time.perf_counter() - self.start_time` counts up to 1.0 and then resets to 0. I'm trying out a few alternative ways now.

Reply:
I left the counter in for simplicity but I could instead create another variable to store the time (one that's a persistent `self` object).

Reply:
Sounds like you left in the `self.start()` call which resets the timer.